### DISCUSSION TITLE
PhotoPrism Instalation - Linux/Docker Compose or direct on NAS

### DISCUSSION THREAD

**Discussion by ziggysid** [photoprism/photoprism#4583] [General]
Hi, been looking for a self hosted photo gallery to replace Google and PhotoPrism looks ideal. Wondering on which way to set it up as I believe there maybe two options that could work for me. 
We are a small family, sharing our photos from android devices and placed on our home NAS (WD EX2Ultra). 
I'd been looking to setup PhotoPrism on Docker Compose, on an old laptop running Lubuntu (24.04.1 LTS), and pointing at the NAS.
Alternatively I understand it may be possible to setup PhotoPrism directly on the NAS.
No previous experience of using Docker Compose (or Linux)
Any thoughts or suggestions as to which route may be better?

**Comment by doobes** [comment]
I just today stood up PP using docker compose in a VM on an HP Elitedesk 705 G3 running Proxmox.

The only challenge that I had was dealing with the Ubuntu/Debian and Broadcom BCM5762 issue. Once that was resolved it was easy.

I'd be less than excited about running PP in a Docker Container on a NAS due to system utilization. I've got 2 cores and 4Gb of memory allocated to PP.

I fired up the interface, and initiated a scan of my folders.  That was 2+ hours ago, the CPU utilization has been ~80% with 90% of the memory used since it started.  And the scan is prolly <10% done. I've got some 70k images in my archive on my Synology NAS.

Would it be faster if PP was running on the NAS, prolly. But I would worry about slowing other things down in the NAS.

Try it on your laptop. It's not hard to set up. 

I used this: https://docs.photoprism.app/getting-started/docker-compose/

And downloaded the docker-compose file.

I would map your the appropriate NAS share to you laptop and then edit the compose appropriately.

Good luck
